### PR TITLE
Update README and remove Foo files

### PR DESCRIPTION
**Pull request by chapulina**
## Summary

Some tweaks to the README and remove temp Foo files now that we have real code.

We can't completely remove `Foo.cc` because `ign-cmake` expects at least one source file.

## Checklist
- [x] Signed all commits for DCO
- [x] ~~Added~~ Removed tests
- [x] Updated documentation (as needed)
- [ ] ~~Updated migration guide (as needed)~~
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

**Comment by azeey** [comment]
`INTEGRATION_versioned_symbols` is failing on Actions. Windows CI is also failing because it's missing `ignition-utils1.lib`. Maybe we can't remove `foo()` after all?

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,3 @@
-# Collect source files into the "sources" variable and unit test files into the
-# "gtest_sources" variable
 ign_get_libsources_and_unittests(sources gtest_sources)
 
-# Create the library target
 ign_create_core_library(SOURCES ${sources} CXX_STANDARD ${c++standard})
```
Comment (scpeters) [inline src/CMakeLists.txt:3]: can we get rid of `SOURCES` if we declare it as `INTERFACE`?

**Comment by scpeters** [comment]
> `INTEGRATION_versioned_symbols` is failing on Actions. Windows CI is also failing because it's missing `ignition-utils1.lib`. Maybe we can't remove `foo()` after all?

I think we would have to disable that test

**Comment by mjcarroll** [comment]
I removed the header from the public `include` folder.  This will now still build a `libignition-utils.so`, but it won't have any meaningful content.  This should allow us the flexibility to remove `foo` without any sort of tick-tock.

File: src/CMakeLists.txt (hunk 1 above)
Comment (chapulina) [reply src/CMakeLists.txt:3]: humm I see that's what `ign-cmake` uses, we could give it a try

Comment (scpeters) [reply src/CMakeLists.txt:3]: I would be fine with it either way

**Review by scpeters — APPROVED** [review]

**Pull request merged**